caffe/src/caffe/layers/molgrid_data_layer.cpp

```diff
@@ -894,6 +1046,11 @@ void MolGridDataLayer<Dtype>::set_grid_ex(Dtype *data, const MolGridDataLayer<Dt
     } else {
         set_grid_minfo(data, recmolcache[ex.receptor], ligmolcache[ligand], transform, peturb, gpu);
     }
+    if (dream) {
+      if (strcmp(ex.ligands[0], "none")==0)
+        ligmolcache[ex.ligands[0]].center = recmolcache[ex.receptor].center;
+      grid_centers[std::string(ex.receptor) + std::string(ex.ligands[0])] = ligmolcache[ex.ligands[0]].center;
```

Comment:
not crazy about grid centers -- centers will change during training with randomness - generaly isn't something you need to store for the entire training set, just the current batch